api/dataset.go

```diff
@@ -386,14 +412,27 @@ func (api *DatasetAPI) deleteDataset(w http.ResponseWriter, r *http.Request) {
 	log.Event(ctx, "delete dataset", log.INFO, logData)
 }
 
+// utility function to cut a slice according to the provided offset and limit.
+// limit=0 means no limit, and values higher than the slice length are ignored
```

Comment:
I would argue limit field should be made a pointer (*int) and therefore 0 value means 0 and would return metadata for the list endpoint.

If the limit field is nil (because it can be due to it being a pointer) than it means use the default value.